src/main/java/com/iota/iri/service/API.java

```diff
@@ -912,4 +817,218 @@ public void shutDown() {
             server.stop();
         }
     }
+
+    // TransactionViewModel, ApproveeViewModel, AddressViewModel, BundleViewModel, TagViewModel Interface
+    private TransactionViewModel transactionViewModelFromHash(final Tangle tangle, final Hash hash) {
+        try {
+            return TransactionViewModel.fromHash(tangle, hash);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    private ApproveeViewModel approveeViewModelGetApprovers(final TransactionViewModel transactionViewModel, final Tangle tangle) {
+        try {
+            return transactionViewModel.getApprovers(tangle);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    private boolean transactionViewModelExist(final Tangle tangle, final Hash hash) {
+        try {
+            return TransactionViewModel.exists(tangle, hash);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    private AddressViewModel addressViewModelLoad(final Tangle tangle, final Indexable indexable) {
+        try {
+            return AddressViewModel.load(tangle, indexable);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    private BundleViewModel bundleViewModelLoad(final Tangle tangle, final Indexable indexable) {
+        try {
+            return BundleViewModel.load(tangle, indexable);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    private TagViewModel tagViewModelLoad(final Tangle tangle, final Indexable indexable) {
+        try {
+            return TagViewModel.load(tangle, indexable);
+        } catch (Exception e) {
+            throw new IllegalStateException(e.getMessage());
+        }
+    }
+
+    /**
+     * Gets a valid uri{@link URI} from an uri String
+     *
+     * @param uriString uri String
+     * @return valid uri
+     * @throws IllegalArgumentException if scheme is not udp or tcp
+     * @throws IllegalArgumentException if uriString has invalid sintax
+     */
+    private URI getValidUri(final String uriString) {
+        try {
+            final URI uri = new URI(uriString);
+            if (!"udp".equals(uri.getScheme()) && !"tcp".equals(uri.getScheme())) {
+                throw new IllegalArgumentException("Invalid uri scheme");
```

Comment:
oh I see.

Reply:
This is an error in the input data. We just have to rise the `IllegalArgumentException` and then send a bad request response.